Content.Server/GameObjects/EntitySystems/PointingSystem.cs

```diff
@@ -98,7 +99,7 @@ public bool TryPoint(ICommonSession? session, GridCoordinates coords, EntityUid
             }
 
             var diff = coords.ToMapPos(_mapManager) - player.Transform.MapPosition.Position;
-            if (diff.LengthSquared > 0.01f)
+            if (diff.LengthSquared > 0.01f && ActionBlockerSystem.CanChangeDirection(player))
```

Comment:
Wrap the whole diff in the ActionBlocker check so it doesn't need to calculate distance at all if CanChangeDirection is false.